crates/cubecl-core/src/frontend/debug.rs

```diff
@@ -23,6 +28,37 @@ pub fn debug_call_expand<C>(
     }
 }
 
+/// Calls an intrinsic op and inserts debug symbols if debug is enabled.
+#[track_caller]
+pub fn spanned_expand<C>(
+    context: &mut CubeContext,
+    line: u32,
+    col: u32,
+    call: impl FnOnce(&mut CubeContext) -> C,
+) -> C {
+    if context.debug_enabled {
+        context.register(DebugInfo::Line { line, col });
+        call(context)
+    } else {
+        call(context)
+    }
+}
+
+/// Adds source instruction if debug is enabled
+#[track_caller]
+pub fn debug_source_expand(context: &mut CubeContext, name: &str, file: &str, line: u32, col: u32) {
+    if context.debug_enabled {
+        // Normalize to linux separators
+        let file = file.replace("\\", "/");
```

Comment:
Can we create a path buf instead?

Reply:
I looked this up and apparently Rust has no way to normalize path separators on pathbuf to a specific one, it can only canonicalize a path to an absolute one for the current system (with the current system's separators